File: analyze_excel.js
const XLSX = require('xlsx');
const path = require('path');

const filePath = path.join(__dirname, 'Tài liệu', 'CT-Mr Mậu Yên Nghĩa.xlsx');

console.log('Reading file:', filePath);

try {
    const wb = XLSX.readFile(filePath);
    console.log('\n==============================');
    console.log('FILE ANALYSIS REPORT');
    console.log('==============================');
    console.log('Total Sheets:', wb.SheetNames.length);
    console.log('Sheet Names:', wb.SheetNames.join(', '));

    wb.SheetNames.forEach(sheetName => {
        console.log('\n==============================');
        console.log('SHEET:', sheetName);
        console.log('==============================');

        const ws = wb.Sheets[sheetName];
        const data = XLSX.utils.sheet_to_json(ws, { header: 1, defval: '' });

        console.log('Total Rows:', data.length);

        // Print first 100 rows
        const maxRows = Math.min(data.length, 100);
        for (let i = 0; i < maxRows; i++) {
            const row = data[i];
            if (row && row.some(cell => cell !== '')) {
                const rowStr = row.map((cell, idx) => {
                    let val = String(cell || '').trim();
                    if (val.length > 30) val = val.substring(0, 27) + '...';
                    return val;
                }).join(' | ');
                console.log(`Row ${i + 1}: ${rowStr}`);
            }
        }
    });
} catch (error) {
    console.error('Error reading file:', error.message);
}

File: analyze_excel.py
import pandas as pd
import sys

file_path = r'd:\ViralWindow_Phan_Mem_Nhom_Kinh\Tài liệu\CT-Mr Mậu Yên Nghĩa.xlsx'

try:
    # Read all sheets
    xlsx = pd.ExcelFile(file_path, engine='openpyxl')
    print(f"File: {file_path}")
    print(f"Number of sheets: {len(xlsx.sheet_names)}")
    print(f"Sheet names: {xlsx.sheet_names}")
    print("=" * 80)
    
    for sheet_name in xlsx.sheet_names:
        df = pd.read_excel(xlsx, sheet_name=sheet_name, header=None)
        print(f"\n{'=' * 80}")
        print(f"SHEET: {sheet_name}")
        print(f"Dimensions: {df.shape[0]} rows x {df.shape[1]} columns")
        print("=" * 80)
        
        # Print first 80 rows
        pd.set_option('display.max_columns', None)
        pd.set_option('display.width', None)
        pd.set_option('display.max_colwidth', 50)
        print(df.head(80).to_string())
        print("\n")
        
except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
